src/data_hub/api.py

```diff
@@ -358,11 +358,81 @@ def upload_file(
     return response.file
 
 
+@beartype
+def add_database_rows(
+    *,
+    database_id: str,
+    data: dict,
+    client=None,
+    _stash: bool = False,
+) -> list[str]:
+    """Add new data to a database.
+
+    Use this function to add new rows, or fragments of rows, to a Deep Origin database.
+
+    Args:
+        database_id: Human ID or System ID of the database
+        data: A dictionary where each key is a column name and each value is a list of values. All values should have the same length. Key names should match column names in the database.
+
+    Returns:
+        A list of row IDs
+
+    """
+    # check that dict has columns that make sense
+    db = _api.describe_row(
+        row_id=database_id,
+        client=client,
+        _stash=_stash,
+    )
+
+    col_names = [col.name for col in db.cols]
+
+    for col in data.keys():
+        if col not in col_names:
+            raise DeepOriginException(
+                message=f"Column `{col}` does not exist in database `{database_id}`."
+            )
```

Comment:
The server will also do this check, if you'd prefer to not do it here.

Reply:
i'd rather catch errors early!

Reply:
It is a possible race condition, though fairly unlikely.